File: Hashmap/Isomorphic Strings/Isomorphic.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func isIsomorphic(s, t string) bool {
	if len(s) != len(t) {
		return false
	}
	mapST := make(map[rune]rune)
	mapTS := make(map[rune]rune)
	for i, sch := range s {
		tch := rune(t[i])
		if mapped, exists := mapST[sch]; exists {
			if mapped != tch {
				return false
			}
		} else if _, exists := mapTS[tch]; exists {
			return false
		} else {
			mapST[sch] = tch
			mapTS[tch] = sch
		}
	}
	return true
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	tcStr, _ := reader.ReadString('\n')
	tc, _ := strconv.Atoi(strings.TrimSpace(tcStr))
	for t := 0; t < tc; t++ {
		s, _ := reader.ReadString('\n')
		t, _ := reader.ReadString('\n')
		s = strings.TrimSpace(s)
		t = strings.TrimSpace(t)
		fmt.Println(isIsomorphic(s, t))
	}
}

File: Hashmap/Isomorphic Strings/Isomorphic.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
 public:
  bool isIsomorphic(string s, string t) {
    if (s.length() != t.length()) return false;
    unordered_map<char, char> mapST;
    unordered_map<char, char> mapTS;
    for (int i = 0; i < s.length(); i++) {
      char sch = s[i];
      char tch = t[i];
      if (mapST.count(sch)) {
        if (mapST[sch] != tch) return false;
      } else {
        mapST[sch] = tch;
      }
      if (mapTS.count(tch)) {
        if (mapTS[tch] != sch) return false;
      } else {
        mapTS[tch] = sch;
      }
    }
    return true;
  }
};

int main() {
  Solution obj;
  int tc;
  cin >> tc;
  cin.ignore();
  while (tc-- > 0) {
    string s, t;
    getline(cin, s);
    getline(cin, t);
    if (obj.isIsomorphic(s, t)) {
      cout << "true" << endl;
    } else {
      cout << "false" << endl;
    }
  }
  return 0;
}
